producer-c/producer-cloudwatch-integ/jobs/runner.groovy

```diff
@@ -0,0 +1,251 @@
+import jenkins.model.*
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+
+START_TIMESTAMP = new Date().getTime()
+HAS_ERROR = false
+
+RUNNING_NODES=0
+
+def buildProducer() {
+  sh  """
+    cd $WORKSPACE/producer-c/producer-cloudwatch-integ && 
+    if [ ! -d "build" ] 
+    then
+        mkdir build
+    fi
+    cd build && 
+    sudo cmake .. && 
+    sudo make 
+  """
+}
+
+def buildConsumer() {
+  sh '''
+    JAVA_HOME='/opt/jdk-13.0.1'
+    PATH="$JAVA_HOME/bin:$PATH"
+    export PATH
+    M2_HOME='/opt/apache-maven-3.6.3'
+    PATH="$M2_HOME/bin:$PATH"
+    export PATH
+    cd $WORKSPACE/consumer-java/aws-kinesis-video-producer-sdk-canary-consumer
+    mvn package
+    touch tmp_jar
+    mvn -Dmdep.outputFile=tmp_jar dependency:build-classpath
+  '''
+}
+
+def runProducer(envs) {
+    def credentials = [
```

Comment:
To be less repetitive in handling credentials, environment variables, and error handling, you can create a wrapper function similar to what webrtc canary has:

https://github.com/aws-samples/amazon-kinesis-video-streams-demos/blob/d163c20b6c0dc49a5247e096507ee01c970728bc/webrtc-c/canary/jobs/runner.groovy#L27-L42